src/board_controller/BrainAlive/BrainAlive.cpp

```diff
@@ -0,0 +1,379 @@
+        if (strcmp (peripheral_address, params.mac_address.c_str ()) == 0)
+        {
+            found = true;
+        }
+    }
+    else
+    {
+        if (!params.serial_number.empty ())
+        {
+            if (strcmp (peripheral_identified, params.serial_number.c_str ()) == 0)
+            {
+                found = true;
+            }
+        }
+        else
+        {
+            if (strncmp (peripheral_identified, "BrainAlive", 10) == 0)
+            {
+                found = true;
+            }
+        }
+    }
+
+    safe_logger (spdlog::level::trace, "address {}", peripheral_address);
+    simpleble_free (peripheral_address);
+    safe_logger (spdlog::level::trace, "identifier {}", peripheral_identified);
+    simpleble_free (peripheral_identified);
+
+    if (found)
+    {
+        {
+            std::lock_guard<std::mutex> lk (m);
+            brainalive_peripheral = peripheral;
+        }
+        cv.notify_one ();
+    }
+    else
+    {
+        simpleble_peripheral_release_handle (peripheral);
+    }
+}
+
+void BrainAlive_Device::Read_Data (simpleble_uuid_t service, simpleble_uuid_t characteristic,
+    uint8_t *data, size_t size, int channel_num)
+{
+    if (size != 46)
+    {
+        safe_logger (spdlog::level::warn, "unknown size of BrainAlive Data {}", size);
+        return;
+    }
+    new_eeg_data[channel_num] = true;
+
+    std::vector<int> eeg_channels = board_descr["eeg_channels"];
+    int sta = (data[1] << 16 | data[2] << 8 | data[3]);
+    int ch1 = ((data[4] << 16 | data[5] << 8 | data[6]) << 8) >> 8;
+    int ch2 = ((data[7] << 16 | data[8] << 8 | data[9]) << 8) >> 8;
+    int ch3 = ((data[10] << 16 | data[11] << 8 | data[12]) << 8) >> 8;
+    int ch4 = ((data[13] << 16 | data[14] << 8 | data[15]) << 8) >> 8;
+    int ch5 = ((data[16] << 16 | data[17] << 8 | data[18]) << 8) >> 8;
+    int ch6 = ((data[19] << 16 | data[20] << 8 | data[21]) << 8) >> 8;
+    int ch7 = ((data[22] << 16 | data[23] << 8 | data[24]) << 8) >> 8;
+    int ch8 = ((data[25] << 16 | data[26] << 8 | data[27]) << 8) >> 8;
+    int IR = ((data[28] << 16 | data[29] << 8 | data[30]) & 0x7FFFF);
+    int RED = ((data[31] << 16 | data[32] << 8 | data[33]) & 0x7FFFF);
+    int Green = ((data[34] << 16 | data[35] << 8 | data[36]) & 0x7FFFF);
+    int X = ((data[37] << 8 | data[38]) << 16) >> 16;
+    int y = ((data[39] << 8 | data[40]) << 16) >> 16;
+    int z = ((data[41] << 8 | data[42]) << 16) >> 16;
+
+ /*   printf ("%d,%d,%d,%d,%d,%d,%d,%d,%d,%d,%d,%d,%d,%d,%d,%d,%d,%d,%d", data[0], sta, ch1, ch2, ch3,
+        ch4, ch5, ch6, ch7, ch8, IR, RED, Green, X, y, z, data[43], data[44], data[45]);
+    printf ("\n");*/
+
+    current_buf[0][eeg_channels[channel_num]] = ch1;
+    current_buf[1][eeg_channels[channel_num]] = ch2;
+    current_buf[2][eeg_channels[channel_num]] = ch3;
+    current_buf[3][eeg_channels[channel_num]] = ch4;
+    current_buf[4][eeg_channels[channel_num]] = ch5;
+    current_buf[5][eeg_channels[channel_num]] = ch6;
```

Comment:
we have a convention that we return data in uV format and for consistency, its better to implement it this way, so lets add a scaler to convert it to uV value